### DISCUSSION TITLE
Statystyki stacji w wykazie wg różnych kryteriów (np. z RDS)

### DISCUSSION THREAD

**Discussion by HyperDX** [TomaszGasior/RadioLista-v3#144] [2. Pomysły Ideas]
Podczas pisania komentarza pod jednym z pomysłów wymyśliłem takie coś jak "Liczba stacji z RDS". Byłoby to wyświetlane obok ogólnej liczby stacji. Zliczałoby to stacje z pełnym RDS oraz te z samym PI.

**Comment by TomaszGasior** [comment]
Sam licznik stacji z RDS wydaje mi się kiepskim pomysłem. Natomiast można to trochę rozszerzyć jako takie ogólne statystyki. Na przykład w formie dymka, który pojawiałby się po najechaniu na aktualny licznik stacji. W dymku znajdowałoby się na przykład:

- liczba stacji z RDS
- liczba stacji z bardzo dobrą jakościa
- liczba stacji niemuzycznych
- liczba „bliskich stacji” (odleglość od nadajnika mniejsza niż ileśtam)
- liczba stacji z Polski (tu bylby problem, bo różnie można Polskę wpisać: Polska, POL, PL itd.)

Obecność pola w dymku mogłaby zależeć od stanu widoczności odpowiedniej kolumny wykazu, bo np. większośc wykazow nie używa rodzaju stacji.

Byłbym wdzięczny za sugestie tego typu „pól statystyk” i ew. krytykę wyżej opisanych.

**Comment by HyperDX** [comment]
Tomku, sam bym na to nie wpadł, świetny pomysł! Ja bym natomiast zrobił te opcje do wyboru. Użytkownik może je odznaczyć lub zaznaczyć w ustawieniach wykazu. Co do liczby stacji z danego kraju... Może ja bym tutaj wszystkie wpisane przez użytkownika opcje w sekcji "Kraj" dał, wtedy by nie było problemu (oczywiście jeśli to wszystko jest możliwe). Opcja z liczbą bliskich stacji, z liczbą stacji niemuzycznych oraz z liczbą bliskich stacji też może być . Co do nowych pól statystyk jak coś przyjdzie mi do głowy to napiszę. Na przykład pole z lokalnością programu może się przydać. Ja z tego nie korzystam, ale może komuś innemu by się przydało. Także liczba stacji odebranych tropo, scatter, Sporadic-E, regular (to przy okazji nowej kolumny).

**Comment by TomaszGasior** [comment]
Ze względu na to iż:

- zrobienie czegoś takiego wymagałoby trochę wysiłku programistycznego — nie byłoby bardzo trudne, ale jednak odrobinę wymagające,
- byłby także wymagany redesign szaty graficznej serwisu (głównie stopki wykazu), żeby to jakoś ładnie przedstawić, 
- nie jest to funkcja niezbędnie krytyczna,
- są ważniejsze rzeczy do zrobienia w RL (oczekujące do dodania kolumny wykazu, język angielski, testy pod czytnikami ekranowymi)

to raczej nieszybko coś takiego wprowadzę. Nie odrzucam takiego pomysłu, natomiast po prostu przesuwam dalej w czasie.

**Comment by TomaszGasior** [comment]
> Może ja bym tutaj wszystkie wpisane przez użytkownika opcje w sekcji "Kraj" dał, wtedy by nie było problemu (oczywiście jeśli to wszystko jest możliwe).

Jest możliwe, ale wymagałoby jeszcze więcej pracy (której aż tak bardzo się nie boję) i mogłoby znacznie pogorszyć czytelnośc takich statystyk (jak ktoś odbiera stacje z kilkunastu krajów na przykład). Jeżeli ktoś naprawdę BARDZO potrzebuje aż tak dokłądnej statystyki wg krajów, solution jest very simpul — pobierz se CSV, zaimportuj w Excelu czy LO Calc i przefiltruj. :)

> Ja bym natomiast zrobił te opcje do wyboru. Użytkownik może je odznaczyć lub zaznaczyć w ustawieniach wykazu.

Za bardzo „pobrudziłoby” to czytelnośc ustawien wykazu. Wydaje mi się dośc odczywiste, że ktoś z wyłączoną kolumną „rodzaj stacji” nie będzie potrzebowac statystyki wg rodzaju stacji, a w wykazie bez kolumny RDS nie będzie mieć sensu statystyka wg obecności RDS.

Ogólnie dobrym podejściem w tworzeniu oprogramowania jest dawanie jak najmniej opcji, jedynie tam, gdzie jest to niezbędne lub bardzo przydtane. Jeśli to tylko możliwe, warto podjąć decyzję za użytkownika i nie zawracać mu tym głowy.

**Comment by HyperDX** [comment]
Co do ilości stacji z danego kraju to nie jest to niezbędne po prostu robie CTRL + F wpisuje kraj i odczytuje ilość znalezionych. Początkowo chciałem tylko, aby była liczba stacji z RDS i tego się trzymam ;).

**Comment by TomaszGasior** [comment]
Tak, jak powiedziałem wyżej, nie zaakceptuję samego licznika RDS. Ew. implementacja mogłaby być wykonana tak, jak pisałem wyżej. Na razie mogę zaproponować ci eksport do pliku CSV i przefiltrowanie arkusza w Excelu/Calcu.
